polar-core/src/parser.rs

```diff
@@ -263,4 +263,19 @@ mod tests {
             }
         ));
     }
+
+    #[test]
+    fn test_primitive_methods() {
```

Comment:
@dhatch noticed this while I was addressing your comments. 

`"abc".startswith("a")` did not previously work.

The fix was to let the LHS of dot be any Exp9 expression (so basically any value, term, or parenthesized expression): https://github.com/osohq/oso/pull/301/commits/6f40f08df25402f22b412c0dcad45ff9a00a9a48#diff-bf4d21bbb0fffe459f1185ea62ce26a4R242